tools/shared/src/main/scala/org/scalajs/core/tools/optimizer/Analyzer.scala

```diff
@@ -349,19 +352,34 @@ final class Analyzer(semantics: Semantics, outputMode: OutputMode,
 
     def instantiated()(implicit from: From): Unit = {
       instantiatedFrom ::= from
-      if (!isInstantiated && isClass) {
+      if (!isInstantiated && (isScalaClass || isAnyRawJSType)) {
         isInstantiated = true
-        accessData()
-        ancestors.foreach(_.subclassInstantiated())
 
-        for ((methodName, from) <- delayedCalls)
-          delayedCallMethod(methodName)(from)
+        if (isScalaClass) {
+          accessData()
+          ancestors.foreach(_.subclassInstantiated())
+
+          for ((methodName, from) <- delayedCalls)
+            delayedCallMethod(methodName)(from)
+        } else {
+          assert(isAnyRawJSType)
+
+          subclassInstantiated()
+
+          if (isJSClass)
+            superClass.instantiated()
+
+          for (methodInfo <- methodInfos.values) {
+            if (methodInfo.isExported)
+              methodInfo.reach(this)(FromExports)
+          }
```

Comment:
Why is this not done in the same place than for normal classes? IIUC, the export reachability for normal classes is actually too aggressive. It reaches even if the class is not instantiated...